File: articles/expected-utility-theory-rational-choice/c/expected_utility.c
#include <math.h>
#include <stdio.h>

double crra_utility(double x, double rho) {
    if (fabs(rho - 1.0) < 1e-8) {
        return log(x);
    }
    return pow(x, 1.0 - rho) / (1.0 - rho);
}

int main(void) {
    double wealth = 50000.0;
    double rho = 1.5;

    double eu_certain = crra_utility(wealth + 100.0, rho);
    double eu_risky = 0.5 * crra_utility(wealth + 40.0, rho) + 0.5 * crra_utility(wealth + 220.0, rho);

    printf("EU certain: %.10f\n", eu_certain);
    printf("EU risky: %.10f\n", eu_risky);
    printf("Choose risky: %s\n", eu_risky > eu_certain ? "yes" : "no");

    return 0;
}

File: articles/expected-utility-theory-rational-choice/go/expected_utility.go
package main

import (
	"fmt"
	"math"
)

func crraUtility(x, rho float64) float64 {
	if math.Abs(rho-1.0) < 1e-8 {
		return math.Log(x)
	}
	return math.Pow(x, 1.0-rho) / (1.0 - rho)
}

func main() {
	wealth := 50000.0
	rho := 1.5

	euCertain := crraUtility(wealth+100.0, rho)
	euRisky := 0.5*crraUtility(wealth+40.0, rho) + 0.5*crraUtility(wealth+220.0, rho)

	fmt.Printf("EU certain: %.10f\n", euCertain)
	fmt.Printf("EU risky: %.10f\n", euRisky)
	if euRisky > euCertain {
		fmt.Println("Choose risky: yes")
	} else {
		fmt.Println("Choose risky: no")
	}
}
